src/metabase/models/login_history.clj

```diff
@@ -10,6 +47,27 @@
     (contains? login-history :session_id) (assoc :active (boolean session-id))
     true                                  (dissoc :session_id)))
 
+(defn- first-login-ever? [{user-id :user_id}]
+  (some-> (db/select [LoginHistory :id] :user_id user-id {:limit 2})
+          count
+          (= 1)))
+
+(defn- first-login-on-this-device? [{user-id :user_id, device-id :device_id}]
+  (some-> (db/select [LoginHistory :id] :user_id user-id, :device_id device-id, {:limit 2})
+          count
+          (= 1)))
```

Comment:
What's the first item that gets returned? I don't understand why it's `{:limit 2} ... (= 1)` instead of `{:limit 1}...empty?`

Reply:
it's just a check to see if there's exactly one instance matching the criteria. With LIMIT 2 we'll get at most 2 results; so count will either be `(= 1)` (it *is* the first instance) or `(= 2)` (it is not the first instance)

Reply:
it seemed more efficient than counting up *all* the items that matched the criteria with `count(*)` ... there could be potentially hundreds.

Reply:
I think @tsmacdonald was thinking if there are no records, it must be the first. But it seems like the implementation creates the record before it checks to see if this is the first login.